core/src/commitment.rs

```diff
@@ -259,7 +280,7 @@ impl AggregateCommitmentService {
                     commitment
```

Comment:
The one thing I'm slightly worried about is that `ancestors` only contains the latest 300 roots (ancestors sourced from status cache here: https://github.com/CriesofCarrots/solana/blob/cde2c1d9baaee528816eeaf515cc8765582a534c/core/src/commitment.rs#L213), meaning any earlier ancestors won't get their commitment stake incremented here, and won't be queryable. 

This might be a problem if we end up in a situation where some group of nodes `A` are consistently more than 300 slots behind the rest of the nodes `B`, but still voting. Then the nodes that are ahead in group `B` won't count the stake of the nodes in group `A` that are behind toward their local commitment.

If we want to solve this I think either we:
1) Keep a lot more roots around in this cache.
2) Instead of mapping `root -> stake confirmed`, map `root range-> stake confirmed`.  For instance if a vote account with stake `S` has root 60, then we store `(0, 60) -> S`. If then another node comes in with stake `S'` and root 30, we store `(0, 30) -> S + S', (30, 60) -> S`, etc.

For instance if a given `vote_state` with root 100 came in